core/silkworm/state/in_memory_state.hpp

```diff
@@ -102,15 +102,15 @@ class InMemoryState : public State {
         storage_;
 
     // block number -> hash -> header
-    std::vector<std::unordered_map<evmc::bytes32, BlockHeader>> headers_;
+    std::map<BlockNum, std::unordered_map<evmc::bytes32, BlockHeader>> headers_;
 
     // block number -> hash -> body
-    std::vector<std::unordered_map<evmc::bytes32, BlockBody>> bodies_;
+    std::map<BlockNum, std::unordered_map<evmc::bytes32, BlockBody>> bodies_;
 
     // block number -> hash -> total difficulty
-    std::vector<std::unordered_map<evmc::bytes32, intx::uint256>> difficulty_;
+    std::map<BlockNum, std::unordered_map<evmc::bytes32, intx::uint256>> difficulty_;
 
-    std::vector<evmc::bytes32> canonical_hashes_;
+    std::map<BlockNum, evmc::bytes32> canonical_hashes_;
```

Comment:
`std::map` is a poor container choice here. There might be a performance drop vs the previous vector. A unordered map would be better.

Reply:
`InMemoryState` is only used for tests, so its performance is not that important.